examples/hf/pippy_gpt2.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+
+# Minimum effort to run this example:
+# $ torchrun --nproc-per-node 4 pippy_gpt2.py
+
+
+import argparse
+import os
+
+import torch
+import torch.distributed as dist
+
+from pippy.IR import Pipe, PipeSplitWrapper, annotate_split_points
+from pippy.PipelineStage import PipelineStage
+
+from transformers import GPT2ForSequenceClassification, GPT2Config
+
+from gen_inputs import generate_inputs_for_model
+
+
+def get_number_of_params(model):
+    return sum(p.numel() for p in model.parameters() if p.requires_grad)
+
+
+def add_split_points(gpt2, nranks):
+    decoders_per_rank = (gpt2.config.n_layer + nranks - 1) // nranks
+    print(f"decoders_per_rank = {decoders_per_rank}")
+    nstages = 1
+    for i in range(1, gpt2.config.n_layer // decoders_per_rank):
+        annotate_split_points(
+            gpt2,
+            {f'transformer.h.{i * decoders_per_rank}': PipeSplitWrapper.SplitPoint.BEGINNING},
```

Comment:
Does this mean the split point is right after this submodule (specified by FQN)?

Reply:
`BEGINNING` means right before; `END` would mean right after.

Reply:
Annotation occurs before tracing. So the module structure is still there at that time.

Reply:
How come there is no corresponding `END`?